src/qibo/gates/measurements.py

```diff
@@ -88,20 +99,25 @@ def __init__(
 
         # list of gates that will be added to the circuit before the
         # measurement, in order to rotate to the given basis
-        if not isinstance(basis, list):
-            basis = len(self.target_qubits) * [basis]
-        elif len(basis) != len(self.target_qubits):
-            raise_error(
-                ValueError,
-                f"Given basis list has length {len(basis)} while "
-                f"we are measuring {len(self.target_qubits)} qubits.",
-            )
         self.basis = []
-        for qubit, basis_cls in zip(self.target_qubits, basis):
+        for qubit, basis_cls in zip(self.target_qubits, self.basis_gates):
             gate = basis_cls(qubit).basis_rotation()
             if gate is not None:
                 self.basis.append(gate)
 
+    @property
+    def raw(self) -> dict:
+        """Serialize to dictionary.
+
+        The values used in the serialization should be compatible with a
+        JSON dump (or any other one supporting a minimal set of scalar
+        types). Though the specific implementation is up to the specific
+        gate.
+        """
+        encoded_simple = super().raw
+        encoded_simple.update(self.result.raw)
```

Comment:
Here you're "flattening" the dictionary, since what was part of `.result` is now serialized together with the other attributes of the measurement gate.

Reply:
Yep, I agree on this one as well, but since at the moment the serialization of a `MeasurementResult` is just the `samples`, building the dict as:
```
{
...
...

"measurement_result": {"samples": [0,1,0,...]}
}
```
looked a bit superfluous.

Reply:
If that's how you will deserialize, that's even fine. That dictionary should not be consumed manually, but automatically. So, if you have a one attribute class, it should be serialized as a one item dictionary - and it is convenient to do that, since you will need to treat fewer special cases.

Otherwise, you can always consider dropping the one attribute class, and make its methods plain functions (or methods of a parent class, if there is a specific one that is always the only one wrapping that).